### PR TITLE
feat: create win UI

### LINKED ISSUE #13: Create UI
Create end-screen containing:
- Winner
- Restart button

### PR DESCRIPTION
**Pull request by 201flaviosilva**
## Summary

- Created a simple and basic UI that appears, with a fade in, as a player is killed;
- The UI as the player killed name and a restart button;

### A illustrative image

Captura de ecrã 2022-11-28 232219

#13

### CODE REVIEW

File: Scripts/Tank.gd [hunk 1]
```diff
@@ -71,6 +74,8 @@ func _on_shoot_timeout():
 	_can_shoot = true
 
 func _on_killed() -> void:
+	GameManager.gameOver(playerName)
```
Comment (Goncalo-Marques) [inline Scripts/Tank.gd]: To guarantee that the tank is destroyed before the `game over` logic is run, I would call this only at the end of the func.
What do you think?

File: Scripts/RestartMenu.gd [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+func _ready() -> void:
+	hide()
+	GameManager.connect("gameOver", self, "gameOver")
+
+func gameOver(playerkilled: String) -> void:
+	print("playerkilled -> ", playerkilled)
+	show()
+	$Background/Label.text = playerkilled + " is Dead!"
+	$AnimationPlayer.play("show")
```
Comment (Goncalo-Marques) [inline Scripts/RestartMenu.gd:10]: The animation looks cool 😄

File: Scripts/RestartMenu.gd [hunk 3]
```diff
@@ -0,0 +1,14 @@
+extends CanvasLayer
+
+func _ready() -> void:
+	hide()
+	GameManager.connect("gameOver", self, "gameOver")
+
+func gameOver(playerkilled: String) -> void:
+	print("playerkilled -> ", playerkilled)
```
Comment (Goncalo-Marques) [inline Scripts/RestartMenu.gd]: Is this really needed?
It looks like a leftover from an early stage😅

File: Scripts/RestartMenu.gd [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+	show()
+	$Background/Label.text = playerkilled + " is Dead!"
+	$AnimationPlayer.play("show")
+
+func _on_RestartBTN_button_down() -> void:
```
Comment (Goncalo-Marques) [inline Scripts/RestartMenu.gd]: to keep consistency, can we rename this: 
```suggestion
func _on_restart_button_down() -> void:
```

**Review by Goncalo-Marques — COMMENTED** [review]
Looks good😄

**Review by Goncalo-Marques — COMMENTED** [review]
Looks good😄

File: Scripts/RestartMenu.gd (hunk 2 above)
Comment (201flaviosilva) [reply Scripts/RestartMenu.gd:10]: thx 😁

File: Scripts/RestartMenu.gd (hunk 3 above)
Comment (201flaviosilva) [reply Scripts/RestartMenu.gd]: true

File: Scripts/RestartMenu.gd (hunk 4 above)
Comment (201flaviosilva) [reply Scripts/RestartMenu.gd]: yap, it's ok, make sense

File: Scripts/Tank.gd (hunk 1 above)
Comment (201flaviosilva) [reply Scripts/Tank.gd]: Yap, I think

**Review by Goncalo-Marques — APPROVED** [review]

**Pull request merged**